metadata/src/main/java/org/apache/kafka/metadata/BrokerState.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kafka.metadata;
+
+import org.apache.kafka.common.annotation.InterfaceStability;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * The broker state.
+ *
+ * The numeric values used here are part of Kafka's public API.  They appear in metrics,
+ * and are also sent over the wire in some cases.
+ *
+ * The expected state transitions are:
+ *
+ * NOT_RUNNING
+ *     ↓
+ * STARTING
+ *     ↓
+ * RECOVERY
+ *     ↓
+ * RUNNING
+ *     ↓
+ * PENDING_CONTROLLED_SHUTDOWN
+ *     ↓
+ * SHUTTING_DOWN
+ */
+@InterfaceStability.Evolving
+public enum BrokerState {
```

Comment:
Not that it does any harm, but is this class needed for this PR? Maybe these values are referenced by the metadata records?

Reply:
I was originally going to remove BrokerStates.scala and replace it with this BrokerState.java file.  That was back when KIP-631 serialized broker states in the RPCs.  But now that we're not doing that any more, there isn't really a strong reason to do this.  So let's just skip it, I guess...